src/main/java/teammates/ui/webapi/GetCourseJoinStatusAction.java

```diff
@@ -38,29 +41,47 @@ public JsonResult execute() {
     }
 
     private JsonResult getStudentJoinStatus(String regkey) {
-        StudentAttributes student = logic.getStudentForRegistrationKey(regkey);
-        if (student == null) {
-            throw new EntityNotFoundException("No student with given registration key: " + regkey);
+        StudentAttributes studentAttributes = logic.getStudentForRegistrationKey(regkey);
+        if (studentAttributes == null || isCourseMigrated(studentAttributes.getCourse())) {
+            Student student = sqlLogic.getStudentByRegistrationKey(regkey);
+
+            if (student == null) {
+                throw new EntityNotFoundException("No student with given registration key: " + regkey);
+            }
+
+            return getJoinStatusResult(student.isRegistered());
         }
-        return getJoinStatusResult(student.isRegistered());
+        return getJoinStatusResult(studentAttributes.isRegistered());
     }
 
     private JsonResult getInstructorJoinStatus(String regkey, boolean isCreatingAccount) {
         if (isCreatingAccount) {
             AccountRequestAttributes accountRequest = logic.getAccountRequestForRegistrationKey(regkey);
-            if (accountRequest == null) {
+            AccountRequest sqlAccountRequest = sqlLogic.getAccountRequestByRegistrationKey(regkey);
+            if (accountRequest == null && sqlAccountRequest == null) {
                 throw new EntityNotFoundException("No account request with given registration key: " + regkey);
             }
-            return getJoinStatusResult(accountRequest.getRegisteredAt() != null);
+
+            if (sqlAccountRequest != null) {
+                return getJoinStatusResult(sqlAccountRequest.getRegisteredAt() != null);
+            }
+            if (accountRequest != null) {
+                return getJoinStatusResult(accountRequest.getRegisteredAt() != null);
+            }
         }
 
-        InstructorAttributes instructor = logic.getInstructorForRegistrationKey(regkey);
+        InstructorAttributes instructorAttributes = logic.getInstructorForRegistrationKey(regkey);
 
-        if (instructor == null) {
-            throw new EntityNotFoundException("No instructor with given registration key: " + regkey);
-        }
+        if (instructorAttributes == null || isCourseMigrated(instructorAttributes.getCourseId())) {
+            Instructor instructor = sqlLogic.getInstructorByRegistrationKey(regkey);
 
-        return getJoinStatusResult(instructor.isRegistered());
+            if (instructor == null) {
+                throw new EntityNotFoundException("No instructor with given registration key: " + regkey);
+            }
+
+            return getJoinStatusResult(instructor.isRegistered());
+        }
```

Comment:
This works, but the logic is abit more complicated and will make removing of the pre-migrated code later on to be more difficult. Will it be better to decouple them and do something similar to lines 60-70? Or even better, you can do something like:

```
if (!isCourseMigrated(...)) {
   // Do datastore things 
} else {
   // Do sql sthings
}
```

Reply:
That makes sense! I have made the changes